web3/utils/contracts.py

```diff
@@ -164,24 +172,32 @@ def prepare_transaction(abi,
         fn_name,
         fn_args,
         fn_kwargs,
+        fn_is_fallback
     )
     return prepared_transaction
 
 
-def encode_transaction_data(abi, web3, fn_name, args=None, kwargs=None):
+def encode_transaction_data(abi, web3, fn_name, args=None, kwargs=None, is_fallback_fn=False):
     fn_abi, fn_selector, fn_arguments = get_function_info(
-        abi, fn_name, args, kwargs,
+        abi, fn_name, args, kwargs, is_fallback_fn
```

Comment:
Rather than using `is_fallback_fn` I'm think we could clean this up using the following pattern.

```
class fallback_fn:
    pass


def encode_transaction_data(abi, web3, fn_identifier, args=None, kwargs=None):
    if fn_identifier is fallback_fn:
        fn_abi, fn_selector, fn_arguments = get_fallback_function_info(...)
    elif is_text(fn_identifier):
        fn_abi, fn_selector, fn_arguments = get_function_info(...)
    else:
        raise TypeError("Unsupported function identifier")

    ...
```

The *new* `get_fallback_function_info` handles all the special cases that were previously handled by `is_fallback` and the `get_function_info` remains simple with less logic forks.

The `fallback_fn` class simply serves as an unambiguous identifier for when we want to specify the fallback function, removing the need for the *extra* flag as well as the need to validate that we aren't passing in both `fn_name` and `is_fallback_fn == True`.

Does this make sense?

Reply:
Yeah, I agree this is much cleaner.

Reply:
Will make the change according to your suggestion.